usaspending_api/common/tracing.py

```diff
@@ -1,110 +1,101 @@
 """
 Module for Application Performance Monitoring and distributed tracing tools and helpers.
 
-Specifically leveraging the Datadog tracing client.
+Specifically leveraging the Grafana Open Telemetry tracing client.
 """
+# Standard library imports
 import logging
 
-from ddtrace import tracer
-from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
-from ddtrace.ext import SpanTypes
-from ddtrace.ext.priority import USER_REJECT
-from ddtrace.internal.writer import AgentWriter
-from ddtrace.span import Span
-from typing import Optional, Callable
+# Typing imports
+from typing import Callable, Optional
 
+# OpenTelemetry imports
+from opentelemetry import trace
+from opentelemetry.trace import SpanKind
+from opentelemetry.trace.status import Status, StatusCode
+
+# The tracer provider should only be set up once, typically in the settings or a dedicated setup module
 _logger = logging.getLogger(__name__)
+tracer = trace.get_tracer_provider().get_tracer(__name__)
 
 
 def _activate_trace_filter(filter_class: Callable) -> None:
     if not hasattr(tracer, "_filters"):
-        _logger.warning("Datadog tracer client no longer has attribute '_filters' on which to append a span filter")
+        _logger.warning("OpenTelemetry does not support direct filter activation on tracer")
     else:
         if tracer._filters:
             tracer._filters.append(filter_class())
         else:
             tracer._filters = [filter_class()]
+        tracer.configure(settings={"FILTERS": tracer._filters})
 
 
-class DatadogEagerlyDropTraceFilter:
+class OpenTelemetryEagerlyDropTraceFilter:
     """
     A trace filter that eagerly drops a trace, by filtering it out before sending it on to the Datadog Server API.
     It uses the `self.EAGERLY_DROP_TRACE_KEY` as a sentinel value. If present within any span's tags, the whole
     trace that the span is part of will be filtered out before being sent to the server.
     """
 
+    # we will probably be getting rid of this class once and for all
     EAGERLY_DROP_TRACE_KEY = "EAGERLY_DROP_TRACE"
 
     @classmethod
     def activate(cls):
         _activate_trace_filter(cls)
 
     @classmethod
-    def drop(cls, span: Span):
-        tracer.get_call_context().sampling_priority = USER_REJECT
-        span.set_tag(cls.EAGERLY_DROP_TRACE_KEY, True)
+    def drop(cls, span: trace.Span):
+        span.set_status(Status(StatusCode.ERROR))
+        span.set_attribute(cls.EAGERLY_DROP_TRACE_KEY, True)
 
-    def process_trace(self, trace):
-        """Drop trace if any span tag has tag with key 'EAGERLY_DROP_TRACE'"""
-        return None if any(span.get_tag(self.EAGERLY_DROP_TRACE_KEY) for span in trace) else trace
+    def process_trace(self, span: trace.Span):
```

Comment:
I think this can be left as is. Alot of the code in regards to the filtering is from the old code using DataDog. And we are not actively using any of the filtering code. I left it in there in the off chance that we default back to Datadog.